study-builder/src/main/java/org/broadinstitute/ddp/studybuilder/EmailBuilder.java

```diff
@@ -0,0 +1,213 @@
+        List<EmailInfo> emails = new ArrayList<>();
+        for (var emailCfg : emailCfgs) {
+            EmailInfo email = updateSendGridEmail(emailCfg);
+            if (email != null) {
+                emails.add(email);
+                LOG.info("Updated email {} with templateId={} versionId={}", email.key, email.templateId, email.versionId);
+            }
+        }
+        if (!emails.isEmpty()) {
+            LOG.info("Updated active versions of {} email templates", emails.size());
+        }
+    }
+
+    private EmailInfo updateSendGridEmail(Config emailCfg) {
+        String key = emailCfg.getString("key");
+        String templateId = ConfigUtil.getStrIfPresent(varsCfg, "emails." + key);
+        if (templateId == null) {
+            throw new DDPException("Could not find email template id for email " + key);
+        }
+
+        File file = cfgPath.getParent().resolve(emailCfg.getString("filepath")).toFile();
+        String html = readRawHtmlContent(file);
+        if (emailCfg.getBoolean("render")) {
+            html = renderHtmlContent(file.getName(), html);
+        }
+        String subject = emailCfg.getString("subject");
+
+        var versionResult = sendGrid.getTemplateActiveVersionId(templateId);
+        versionResult.rethrowIfThrown(e -> new DDPException("Error fetching active version for email " + key, e));
+        if (versionResult.hasError()) {
+            LOG.error("Could not find active version for email {} with template id {}: {}",
+                    key, templateId, versionResult.getError());
+            return null;
+        }
+        String versionId = versionResult.getBody();
+
+        // Note: we're not updating version name since we didn't update template name, so we keep them the same.
+        var updateResult = sendGrid.updateTemplateVersion(templateId, versionId, null, subject, html);
+        updateResult.rethrowIfThrown(e -> new DDPException("Error while updating email " + key, e));
+        if (updateResult.hasError()) {
+            LOG.error("Error while updating active version {} for email {} with template id {}: {}",
+                    versionId, key, templateId, versionResult.getError());
+            return null;
+        }
+
+        return new EmailInfo(key, templateId, versionId);
+    }
+
+    private String readRawHtmlContent(File file) {
+        if (!file.exists()) {
+            throw new DDPException("File " + file + " does not exist");
+        }
+        try {
+            return Files.readString(file.toPath());
+        } catch (IOException e) {
+            throw new DDPException(e);
+        }
+    }
+
+    private String renderHtmlContent(String name, String rawHtml) {
+        var ctx = new VelocityContext(varsCfg.root().unwrapped());
+        StringWriter writer = new StringWriter();
+        velocity.evaluate(ctx, writer, name, rawHtml);
+        return writer.toString();
+    }
+
+    private static class EmailInfo {
+        public String key;
```

Comment:
Ctrl-n to generate the getters and make these private?  I know it seems silly when it's clear that this is a struct, but down the road, it might change into something else and then mutability can get confusing.